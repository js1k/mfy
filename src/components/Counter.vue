<template>
    <div>{{count}}</div>
</template>
<script>
export default {
    data () {
        return {
            count: 0
        };
    },
    mounted () {
    },
    methods: {
        inc () {
            this.count++;
        },
        reset () {
            this.count = 0;
        }
    }
};
</script>
<style lang="scss">

</style>
