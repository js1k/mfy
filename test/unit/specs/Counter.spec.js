import { mount } from '@vue/test-utils';
// import Vue from 'vue';
import Counter from '@/components/Counter.vue';

describe('Counter.vue', () => {
    it('渲染Counter组件', () => {
        const wrapper = mount(Counter);
        expect(wrapper.element).toMatchSnapshot();
    });
    it('初始化为0', () => {
        const wrapper = mount(Counter);
        expect(wrapper.vm.count).toEqual(0);
    });
    it('加1', () => {
        const wrapper = mount(Counter);
        wrapper.vm.inc();
        expect(wrapper.vm.count).toEqual(1);
    });
    it('重置', () => {
        const wrapper = mount(Counter);
        wrapper.vm.reset();
        expect(wrapper.vm.count).toEqual(0);
    });
});
