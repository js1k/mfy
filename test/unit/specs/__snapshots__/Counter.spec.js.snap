// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue 渲染Counter组件 1`] = `
<div>
  0
</div>
`;
